### ISSUE TITLE
Build python3 by default?

### ISSUE THREAD

**Issue by rfk** [rfk/pypyjs#60]

**Comment by rfk** [comment]
FWIW I don't feel strongly about this and I feel like python2-by-default will be easier because that's what upstream pypy is shipping.  But folks are welcome to weigh in here with alternative opinions :-)

One option is to track the py3k branch of pypy as a separate git submodule, eg. ./deps/py3k, and add targets to the makefile for building both py2 and py3 easily.

**Comment by techdragon** [comment]
I would say the makefile should be the minimum. No reason to actively avoid having builds of py3.

But personally I think py3 should be the default. We have no baggage at present, no reason to hold back. 3.4 :+1: 3.5 with type hints and all the trimmings :+1: ok... maybe just 'latest pypy3' will suffice.

**Comment by perkinslr** [comment]
Personally, I don't want to lose python2 support.  I wouldn't mind having a python3 build too, but I mildly dislike python3 as a language, and twisted+nevow (which I have running on pypyjs over top of WebSockets) doesn't support python3; twisted itself is only partially runnable on python3.

**Comment by crantila** [comment]
Yes to both! PyPy isn't great with Python 3 yet, so maybe it shouldn't be the default here, but Python 3 will happen sooner or later so we might as well jump aboard.

**Comment by shakna-israel** [comment]
PyPy3 isn't that well supported, and there's a few things like numpy that are... Difficult.

Personally, I love Python 3, and I think support is almost there for most things.

A makefile for both seems the best compromise in my mind.

**Comment by rfk** [comment]
Cool, sounds like there's a pretty solid argument for building both py2 and py3 as different targets in the makefile.  I'll see about cleaning up the hackery I did to get it building for the pycon talk and commit something.

**Comment by jedie** [comment]
:+1:

**Comment by rfk** [comment]
As of https://github.com/pypyjs/pypyjs/commit/073efed7415238f34810db5a23b4b64efbbcebbe there are targets in the Makefile for building a (very experimental) python3 version:

```
> make release3
```

I'll probably keep tweaking the process and maybe rename the target if I can think of something better, but wanted to get the initial version of this committed.  Last time I had this working, I carelessly tossed it all away and had to start over...

**Comment by orionz** [comment]
Is there any way to get access to the pypy3 version without building it yourself?  Any chance someone could package a pypyjs3 npm module?
